src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// The complex Target Framework property consists of three UI-exposed properties: target framework moniker, target platform and target platform version.
+    /// Those combine and produce a single value like this: [target framework alias]-[target platform][target platform version].
+    /// This class intercepts those properties from the UI to saved their values into the TargetFramework property.
+    /// In the case of retrieving the values of those properties, this class looks for the associated properties found in the project's configuration.
+    /// </summary>
+    [ExportInterceptingPropertyValueProvider(
+        new[]
+        {
+            InterceptedTargetFrameworkProperty,
+            TargetPlatformProperty,
+            TargetPlatformVersionProperty
+        },
+        ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class ComplexTargetFrameworkValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private const string InterceptedTargetFrameworkProperty = "InterceptedTargetFramework";
+        private const string TargetPlatformProperty = "TargetPlatform";
+        private const string TargetPlatformVersionProperty = "TargetPlatformVersion";
+        private const string TargetFrameworkProperty = "TargetFramework";
+        private const string MinimumPlatformVersionProperty = "MinimumPlatformVersion";
```

Comment:
Could use the existing constants instead:

- `ConfigurationGeneral.TargetPlatformVersionProperty`
- `ConfigurationGeneral.TargetFrameworkProperty`
- `ConfigurationGeneral.MinimumPlatformVersionProperty`

Using the existing constants makes it easier to find where we're using a given XAML rule property throughout the code (via Find All References).

Reply:
Agree, added in the next commit. Thanks!